src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -19,16 +19,42 @@ const ApplicationLauncher = _.flow(
   Utils.forwardRefWithName('ApplicationLauncher'),
   wrapWorkspace({
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
-    title: _.get('application')
+    title: _.get('application'),
+    activeTab: appLauncherTabName
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ name, refreshRuntimes, runtimes, persistentDisks, application, workspace, workspace: { workspace: { googleProject, bucketName } } }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
   const [busy, setBusy] = useState(false)
+  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') // We've already init Welder if app is Jupyter.
 
   const runtime = getCurrentRuntime(runtimes)
   const runtimeStatus = getConvertedRuntimeStatus(runtime) // preserve null vs undefined
 
+  useEffect(() => {
+    const runtime = getCurrentRuntime(runtimes)
+    const runtimeStatus = getConvertedRuntimeStatus(runtime)
+
+    const setupWelder = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('Error setting up analysis')
+    )(async () => {
+      const localBaseDirectory = ``
+      const localSafeModeBaseDirectory = ``
+      const cloudStorageDirectory = `gs://${bucketName}/notebooks`
+
+      await Ajax()
+        .Runtimes
+        .fileSyncing(googleProject, runtime.runtimeName)
+        .setStorageLinks(localBaseDirectory, localSafeModeBaseDirectory, cloudStorageDirectory, `.*\\.Rmd`)
+    })
+
+    if (!isWelderInit && runtimeStatus === 'Running') {
+      setupWelder()
+      setIsWelderInit(true)
+    }
+  }, [googleProject, name, isWelderInit, runtimes, bucketName])
```

Comment:
The `useEffect` hook would be run one more time unnecessarily when it changes `isWelderInit`, a state that is in its dependency array so we should take it out?

```suggestion
  }, [googleProject, name, runtimes, bucketName])
```

`isWelderInit` is only changed by that `useEffect` so it shouldn't miss any updates to that state due to the removal.

Reply:
Ah that's a remnant of a previous refactor, thanks

Reply:
![image](https://user-images.githubusercontent.com/6465084/134955777-f20d4979-ee90-4ca6-8e8b-fc9fcbe0287e.png)

Reply:
Yup, it should be suppressed in this instance to accurately capture the intent of this line. Add this:
```
// eslint-disable-line react-hooks/exhaustive-deps
```